### PR TITLE
Tooling - Add jest-axe

### LINKED ISSUE #2808: Improve accessibility checks
Going further than #2456's implementation of a11y checks in eslint.

@esizer and I had a quick call to discuss some things in #2633, and realized we both thought there was benefit from more a11y checks.

Eric mentioned that since these tools don't cover all cases, it's good to have multiple coverage, so they catch the most surface area. Shared this article: https://www.adrianbolonio.com/en/accessibility-github-actions/

Prior research: https://github.com/GCTC-NTGC/gc-digital-talent/issues/2456#issuecomment-1097148259 

## Potential To Dos
- github actions run with these tools
  - [x] add eslint axe #2456 
  - [ ] axe (see article below)
  - [ ] axe-linter (see article below)
  - [ ] pa11y (see article below) +1 mgifford
      - https://github.com/pa11y/pa11y-ci
  - [ ] lighthouse (see article below)
  - [ ] cypress axe linting. +1 mgifford
      - https://github.com/component-driven/cypress-axe
  - [ ] add to definition of done http://go.talent.c4nada.ca/done

## Open Questions
1. can we run some of these in the same workflow? if we keep them separate, is it visually noisy to have too many checks in PRs?
2. are some of the above tools totally redundant to use together, even though slightly different?

### PR DESCRIPTION
**Pull request by esizer**
Related to but will not fix #2808 

## Summary

This adds `jest-axe` to each workspace and adds an axe test to each of our `*.test.tsx` files.

## Changes

- Installs `jest-axe`
- Creates a `setup.ts` to extend `expect` with `toHaveNoViolations`
- Adds a "should have no accessibility errors" test to each of our `*.test.tsx` files
- Fixes some minor errors that popped up while testing
  - IAP heading levels
  - Pagination missing label on number input
  - Removes invalid `aria-required` from `fieldset`

## Important Note

It turns out you shouldn't use `aria-required` on a `fieldset` as outlined in the [comment on a related w3c/aria issue](https://github.com/w3c/aria/issues/1322#issuecomment-693667305). It seems to imply that **all** inputs within the `fieldset` are required which is not what we want.

It seems like a good alternative is to remove the `required` attribute entirely and just provide a default. Since, once selected, you cannot deselect a radio it should have the same outcome. This was mentioned by someone who identifies as a screen reader user in a stack overflow comment.

 > "From a usability perspective I prefer to have a default selection since if I see a default I usually skip to the next section of the form assuming the default isn’t obviously wrong. Whether you feel comfortable with providing the user a default choice and not forcing them to make a selection would depend on the importance of the information your application requires and the consequences if that information is wrong."

Ref: https://stackoverflow.com/a/8511841

### CODE REVIEW

**Review by mnigh — CHANGES_REQUESTED** [review]
Nice work. A few things to address but headed in the right direction. Also, were the jest tests failing before you added some new tests in or did these as a result of the new tests?

File: frontend/common/tests/setup.ts [hunk 1]
```diff
@@ -1,3 +1,8 @@
+import { toHaveNoViolations } from "jest-axe";
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/common/tests/setup.ts:1]: The file was already here so I just used it 🤷 

I'll see about moving it.

[the remark this answers is not in the document]
Comment (esizer) [reply frontend/common/tests/setup.ts]: Trickier than I thought, but it was moved in [move common test setup file](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3288/commits/a2ee8b9a61e0096b5670395f5a61f8ca239bb005)

File: frontend/common/src/tests/setup.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { toHaveNoViolations } from "jest-axe";
```
Comment (mnigh) [inline frontend/common/src/tests/setup.ts:1]: Still seeing 'Unable to resolve path to module 'jest-axe' after running refresh_all.sh

Comment (esizer) [reply frontend/common/src/tests/setup.ts:1]: wth, it is def in the [package.json](https://github.com/GCTC-NTGC/gc-digital-talent/blob/a2ee8b9a61e0096b5670395f5a61f8ca239bb005/frontend/common/package.json#L113) 😖

**Review by mnigh — APPROVED** [review]
🚀

**Pull request merged**